### PR TITLE
Fixing multiple issues in notification_center.

### PR DESCRIPTION
**Pull request by aliabbasrizvi**
Summary
-------

`notification_center.NotificationCenter` did not have any unit tests for the class. All the related unit tests were absorbed in tests for `optimizely.Optimizely`. This PR:
- Creates `test_notification_center.py` to introduce unit tests for the `NotificationCenter` class.
- Fixes naming in the `NotificationCenter` class. This includes fixing incorrect API names which have been marked as deprecated and we will discontinue support in the next major release.
- Removes tests in `test_optimizely.py` which were focusing on tests captured in `test_notification_center.py`.

Test plan
---------

- Unit tests

### CODE REVIEW

File: optimizely/notification_center.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2017, Optimizely
+# Copyright 2017-2019, Optimizely
```
Comment (mikeng13) [inline optimizely/notification_center.py]: Has it really been 2 years since we touched this?

File: optimizely/notification_center.py [hunk 2]
```diff
@@ -14,43 +14,52 @@
...
-                filter(lambda tup: tup[1] == notification_callback, self.notifications[notification_type]),
-                0) > 0:
-          return -1
-      self.notifications[notification_type].append((self.notification_id, notification_callback))
+    if notification_type not in NOTIFICATION_TYPES:
+      self.logger.error('Invalid notification_type: {} provided. Not adding listener.'.format(notification_type))
+      return -1
 
-    ret_val = self.notification_id
+    if reduce(lambda a, b: a + 1, filter(lambda tup: tup[1] == notification_callback,
```
Comment (mikeng13) [inline optimizely/notification_center.py]: I don't find this to be very readable, but then again, I am not fluent in Python :)

File: optimizely/notification_center.py (hunk 1 above)
Comment (aliabbasrizvi) [reply optimizely/notification_center.py]: We touched it some time last year briefly, but this was not updated.

File: optimizely/notification_center.py (hunk 2 above)
Comment (aliabbasrizvi) [reply optimizely/notification_center.py]: Let me try and update this as well. This entire thing was not very readable.

**Pull request merged**